src/saga/diff.js

```diff
@@ -143,8 +77,23 @@ async function diff(context, initialCommit, finalCommit) {
         const initialFormulas = await getFormulas(context, modifiedSheetNamePairs[i].initialSheet);
         const finalFormulas = await getFormulas(context, modifiedSheetNamePairs[i].finalSheet);
 
-        const result = simpleDiff2D(initialFormulas, finalFormulas, modifiedSheetNamePairs[i].sheetName);
-        sheetChanges.push(result);
+        const changes = simpleDiff2D(initialFormulas, finalFormulas, modifiedSheetNamePairs[i].sheetName);
+
+        // No changes, mark it as such
+        if (changes.length === 0) {
```

Comment:
What is the benefit of having a no change data type? I ended up removing it given we will just ignore it on the UI. Just extra data to pass around and a case to handle in displaying the UI.